File: CS/15_Document/AddCustomMetaDataFromXml.cs
using System;
using System.IO;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Interchange.Metadata;

namespace AddCustomMetaDataFromXml
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Load the input Pdf file
            PdfDocument doc = new PdfDocument(@"..\..\..\..\..\..\Data\Template_Pdf_4.pdf");

            //Load the xml file of metadata
            Stream stream = new FileStream(@"..\..\..\..\..\..\Data\MetaData.xml", FileMode.Open);

            //Set the metadata from xml file to Pdf file
            doc.Metadata = PdfXmpMetadata.Parse(stream);

            //Save the result Pdf file
            string resultPath = @"CustomMetaFromXml_result.pdf";
            doc.SaveToFile(resultPath);

            //Launch the Pdf file
            PDFDocumentViewer(resultPath);
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB/15_Document/AddCustomMetaDataFromXml.vb
Imports System.IO
Imports Spire.Pdf
Imports Spire.Pdf.Interchange.Metadata

Namespace AddCustomMetaDataFromXml
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfDocument object and load the PDF file from the specified path
			Dim doc As New PdfDocument("..\..\..\..\..\..\Data\Template_Pdf_4.pdf")

			' Open a file stream to read the XML metadata file
			Dim stream As Stream = New FileStream("..\..\..\..\..\..\Data\MetaData.xml", FileMode.Open)

			' Parse the XML metadata using PdfXmpMetadata.Parse() method and assign it to the document's Metadata property
			doc.Metadata = PdfXmpMetadata.Parse(stream)

			' Specify the path for the resulting PDF file
			Dim resultPath As String = "CustomMetaFromXml_result.pdf"

			' Save the modified document to the specified path
			doc.SaveToFile(resultPath)

			' Close the document
			doc.Close()

			' Launch the Pdf file
			PDFDocumentViewer(resultPath)
		End Sub

		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
